src/test/shared/settingsConfiguration.test.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as vscode from 'vscode'
+import { extensionSettingsPrefix } from '../../shared/constants'
+import { DefaultSettingsConfiguration } from '../../shared/settingsConfiguration'
+
+describe('DefaultSettingsConfiguration', () => {
+    // These tests use an actual extension setting, because vscode.WorkspaceConfiguration fails when
+    // you attempt to update one that isn't defined in package.json. We will restore the setting value
+    // at the end of the tests.
```

Comment:
This is an example of global state that can introduce side effects on other tests and prevents us from parallelizing tests.

It would be useful if we could stub the settings somehow.

Reply:
Agreed, but we can address parallel tests if/when we make parallel tests. @bryceitoc9 suggestion for workspace-level settings would negate the need to preserve the value.

This test is closer to an integration test, in that we need to exercise code that uses VS Code's settings system. A stub would not necessarily give us the assurances we'd want.